src/addrman.cpp

```diff
@@ -589,8 +583,7 @@ void AddrManImpl::Good_(const CService& addr, bool test_before_evict, int64_t nT
             m_tried_collisions.insert(nId);
```

Comment:
Above, on line 581 there is a confusing log which says "moving", but it is actually not moving anything - it only adds the ID to a list of collided IDs. The promoted entry remains in the new table (`Good()` fails to promote it to tried) and the existent entry remains in tried.

```diff
     if (test_before_evict && (vvTried[tried_bucket][tried_bucket_pos] != -1)) {
         // Output the entry we'd be colliding with, for debugging purposes
         auto colliding_entry = mapInfo.find(vvTried[tried_bucket][tried_bucket_pos]);
-        LogPrint(BCLog::ADDRMAN, "Collision inserting element into tried table (%s), moving %s to m_tried_collisions=%d\n", colliding_entry != mapInfo.end() ? colliding_entry->second.ToString() : "", addr.ToString(), m_tried_collisions.size());
+        LogPrint(
+            BCLog::ADDRMAN,
+            "Cannot move %s from new to tried due to a collision%s. Collisions=%u.",
+            addr.ToString(),
+            colliding_entry != mapInfo.end() ? " with " + colliding_entry->second.ToString() : "",
+            m_tried_collisions.size());
         if (m_tried_collisions.size() < ADDRMAN_SET_TRIED_COLLISION_SIZE) {
             m_tried_collisions.insert(nId);
         }
     } else {
```